extensions/funqy/funqy-knative-events/runtime/src/main/java/io/quarkus/funqy/runtime/bindings/knative/events/VertxRequestHandler.java

```diff
@@ -95,8 +113,258 @@ public void handle(RoutingContext routingContext) {
         }
     }
 
-    private static final ResponseProcessing NOOP = () -> {
-    };
+    private void processCloudEvent(RoutingContext routingContext) {
+        final HttpServerRequest httpRequest = routingContext.request();
+        final HttpServerResponse httpResponse = routingContext.response();
+        final boolean binaryCE = httpRequest.headers().contains("ce-id");
+
+        httpRequest.bodyHandler(bodyBuff -> executor.execute(() -> {
+            try {
+                final String ceType;
+                final String ceSpecVersion;
+                final JsonNode structuredPayload;
+
+                if (binaryCE) {
+                    ceType = httpRequest.headers().get("ce-type");
+                    ceSpecVersion = httpRequest.headers().get("ce-specversion");
+                    structuredPayload = null;
+                } else {
+                    try {
+                        structuredPayload = mapper.readTree(bodyBuff.getBytes());
+                        ceType = structuredPayload.get("type").asText();
+                        ceSpecVersion = structuredPayload.get("specversion").asText();
+                    } catch (IOException e) {
+                        routingContext.fail(e);
+                        return;
+                    }
+                }
+
+                if (!"1.0".equals(ceSpecVersion) && !"0.3".equals(ceSpecVersion)) {
```

Comment:
@patriot1burke I am checking specversion even before `CloudEvent<>` creation. Anything but `0.3` and `1.0` is a bad request.

Reply:
Maybe I could return 501 Not Implemented with message saying that we support only `0.3` and `1.0`.

Reply:
I don't agree on 501.  Spec will probably be backward compatible.